### ISSUE TITLE
[core] jadx file.jar throws "classes.dex, error: Unexpected magic"

### ISSUE THREAD

**Issue by Samulia-Sanchez** [skylot/jadx#1521]
I am trying to decompile the attached .jar file with jadx CLI, but get the error below. 
jadx-gui opens the file without issues.

**Describe error**
```
> jadx .\sm.jar
INFO  - output directory: sm
INFO  - loading ...
ERROR - Failed to load file: classes.dex, error: Unexpected magic: [0x64, 0x65, 0x78, 0x0a, 0x30, 0x33, 0x39, 0x00]
com.android.dex.DexException: Unexpected magic: [0x64, 0x65, 0x78, 0x0a, 0x30, 0x33, 0x39, 0x00]
        at com.android.dex.TableOfContents.readHeader(TableOfContents.java:86)
        at com.android.dex.TableOfContents.readFrom(TableOfContents.java:72)
        at com.android.dex.Dex.loadFrom(Dex.java:142)
        at com.android.dex.Dex.<init>(Dex.java:98)
        at jadx.core.utils.files.InputFile.makeDexBuf(InputFile.java:150)
        at jadx.core.utils.files.InputFile.loadFromZip(InputFile.java:109)
        at jadx.core.utils.files.InputFile.searchDexFiles(InputFile.java:59)
        at jadx.core.utils.files.InputFile.addFilesFrom(InputFile.java:36)
        at jadx.api.JadxDecompiler.loadFiles(JadxDecompiler.java:124)
        at jadx.api.JadxDecompiler.load(JadxDecompiler.java:87)
        at jadx.cli.JadxCLI.processAndSave(JadxCLI.java:30)
        at jadx.cli.JadxCLI.main(JadxCLI.java:17)
WARN  - Unknown 'R' class, create references to 'R'
INFO  - processing ...
INFO  - done
```

[sm.zip](https://github.com/skylot/jadx/files/8872593/sm.zip)

**Comment by skylot** [comment]
Looks like you are using a very old version (like v1.1.0 which released in 2019).
Latest version works fine with your sample, please upgrade jadx :slightly_smiling_face:

**Comment by Samulia-Sanchez** [comment]
> Looks like you are using a very old version (like v1.1.0 which released in 2019). Latest version works fine with your sample, please upgrade jadx 🙂

You are right. Thank you for pointing me in the right direction. I've been using jadx-gui for a long time, but tried using jadx for the first time, not realizing it's a separate tool I had to download and configure.
Somehow I had an old jadx version in the PATH and assumed it was the same version as jadx-gui. 
Investigation showed that it came with Android Development Kit (ADK), which I tested a long time ago and forgot about.

I have now removed the old version, and downloaded the cross-platform cli and gui bundle and added it to the PATH, and its all good now.

Thank you for helping me, and sorry for taking up your time with such a basic mistake.
